FlexConfirmMail/Thunderbird
# Add Native Messaging Host for macOS

# Which issue(s) this PR fixes:

N/A

# What this PR does / why we need it:

This adds Native Messaging Host for macOS, including changes to separate platform-specific codes and platform-independent codes.

# How to verify the fixed issue:

Build the binary and pack it on macOS.

## The steps to verify:

1. Prepare Linux environment to build binaries.
2. Run `cd webextensions && make native-messaging-host`.
3. Confirm that a directory `webextensions/native-messaging-host/darwin` is created.
4. Copy the directory `webextensions/native-messaging-host/darwin` to a macOS environment.
5. On the macOS environment, run `build_pkg.sh`.
6. Confirm that a file `com.clear_code.flexible_confirm_mail_we_host.pkg` is created.
7. Right-click or Control-click on the pkg file and choose "Open", and continue to install.
8. Start Thunderbird.
9. Install FlexConfirmMail for Thunderbird.
10. Open the options page of FlexConfirmMail.
11. Click "Choose..." button beside any file path input form under the "extra rules" section.

## Expected result:

A file chooser dialog appears.
